### ISSUE TITLE
reproducing fine-tuned tokens. (textual inversion)

### ISSUE THREAD

**Issue by bansh123** [brandontrabucco/da-fusion#24]
I have attempted to reproduce the results of the few-shot classification task on the PASCAL VOC dataset.
I managed to achieve comparable outcomes when utilizing the fine-tuned tokens you previously shared via the Google Drive link.
However, I was unsuccessful in reproducing the fine-tuned tokens.
When employing fine_tune.py and aggregate_embeddings.py with the provided scripts, I obtained inferior tokens, resulting in significantly lower accuracy (approximately a 10% gap in 1-shot).

Am I overlooking something?

**Comment by brandontrabucco** [comment]
Hello [bansh123](https://github.com/bansh123),

Thanks for bringing this to my attention, I'll revisit this experiment to determine why you are seeing this behavior.

In the meantime, for the latest paper update on OpenReview, we used fine_tune_upstream.py, using the class name of the object as the initialization, and training 4 vectors per concept in the dataset with the `--num_vectors` argument.

We trained these tokens using a batch size of 8, stable diffusion 1.4, and 2000 gradient descent steps, a learning rate of 5.0e-04, scaled by the effective batch size via the `--scale_lr` argument to the script.

Which figure are you working to reproduce?

-Brandon

**Comment by bansh123** [comment]
Thank you for your response. 
I am currently working on reproducing Figure 5, and I plan to tackle Figure 8 subsequently.
I will conduct the experiments again, adhering to your configuration. Thank you!

**Comment by jsw6872** [comment]
> I have attempted to reproduce the results of the few-shot classification task on the PASCAL VOC dataset. I managed to achieve comparable outcomes when utilizing the fine-tuned tokens you previously shared via the Google Drive link. However, I was unsuccessful in reproducing the fine-tuned tokens. When employing fine_tune.py and aggregate_embeddings.py with the provided scripts, I obtained inferior tokens, resulting in significantly lower accuracy (approximately a 10% gap in 1-shot).
> 
> Am I overlooking something?

I've tried using the pascal token provided by the author, but I can't reproduce the performance. 
If you don't mind me asking, could you share the seed or hyperparameters you used?

![visualization_pascal_overall](https://github.com/brandontrabucco/da-fusion/assets/94580562/f5d8f130-9ef9-44c9-8ef0-acb1023339aa)

Thank you.

**Comment by zhixiongzh** [comment]
Hi @brandontrabucco,

is it possible to share in google drive the new embeddings, which you get by initializing with the original class name and 4 num_vectors? Recomputing the embedding needs a lot of compute resources for more classes.

Thanks in advance for considering this request.

**Comment by zhixiongzh** [comment]
@brandontrabucco 
another question, how to ensure no `ValueError` when using the class name of the object as the initialization, since you have following check in the `fine_tune_upstream.py`
```
    if len(token_ids) > 1:
        raise ValueError("The initializer token must be a single token.")
```
some original class names which have multiple words would have more tokens than 1, e.g. pink primrose in Flowers102 dataset.

**Comment by tanriverdiege** [comment]
> Thank you for your response. I am currently working on reproducing Figure 5, and I plan to tackle Figure 8 subsequently. I will conduct the experiments again, adhering to your configuration. Thank you!

@bansh123 Did you reproduce the results, I am having trouble getting the results presented in the paper (Figure 5 to be specific). Do you have any tips (scripts) that you can share to help me ? Thank you.

**Comment by tanriverdiege** [comment]
> > I have attempted to reproduce the results of the few-shot classification task on the PASCAL VOC dataset. I managed to achieve comparable outcomes when utilizing the fine-tuned tokens you previously shared via the Google Drive link. However, I was unsuccessful in reproducing the fine-tuned tokens. When employing fine_tune.py and aggregate_embeddings.py with the provided scripts, I obtained inferior tokens, resulting in significantly lower accuracy (approximately a 10% gap in 1-shot).
> > Am I overlooking something?
> 
> I've tried using the pascal token provided by the author, but I can't reproduce the performance. If you don't mind me asking, could you share the seed or hyperparameters you used?
> 
> ![visualization_pascal_overall](https://private-user-images.githubusercontent.com/94580562/315214336-f5d8f130-9ef9-44c9-8ef0-acb1023339aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2NTAxNzYsIm5iZiI6MTcyMDY0OTg3NiwicGF0aCI6Ii85NDU4MDU2Mi8zMTUyMTQzMzYtZjVkOGYxMzAtOWVmOS00NGM5LThlZjAtYWNiMTAyMzMzOWFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDIyMTc1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0ZWY3ZDU1OTQ3M2Y4YmViMzhlYzYxYWIwOWFkNzFjYTNlMWZiZTI4ZTYxOTI2NjMzMmE3YTRhNzA4NjZiN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-zXGBYav3oWbNF8fK1M7buzxckiAMhOPQwwolufbNvk)
> 
> Thank you.

@jsw6872 Did you manage to reproduce the results ?

**Comment by tanriverdiege** [comment]
> @brandontrabucco another question, how to ensure no `ValueError` when using the class name of the object as the initialization, since you have following check in the `fine_tune_upstream.py`
> 
> ```
>     if len(token_ids) > 1:
>         raise ValueError("The initializer token must be a single token.")
> ```
> 
> some original class names which have multiple words would have more tokens than 1, e.g. pink primrose in Flowers102 dataset.

How did you resolve the issue ? I would really appreciate your help. Thanks !
